test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/StorageBindingTests.cs

```diff
@@ -257,6 +259,51 @@ public static IHostBuilder ConfigureGeneratedFunctionMetadataProvider(this IHost
                     expectedGeneratedFileName,
                     expectedOutput);
             }
+
+            [Fact]
+            public async void TestInvalidBlobCardinalityMany()
```

Comment:
Can we add another test for IsBatched false but with List<string>? I would expect that to fail as well.

And a valid scenario test would also be good to have e.g.

IsBatched = false or not set / string -> no error
IsBatched = true / IList<string> - no error

Reply:
Both of those valid scenarios for Blob were captured in Surbhi's PR (test cases above in this same file). I wrote some but erased them after I realized they were already there the next day xD. Similar scenarios are also completely captured in the EH tests.

I haven't written tests for the scenario of `IsBatched = false` with an iterable collection because I don't believe the legacy version checks for that (or has tests for it). [Here](https://github.com/Azure/azure-functions-dotnet-worker/blob/0fd7bf6aef005e4b8a14874506bf7a8ad7ad73ef/sdk/Sdk/FunctionMetadataGenerator.cs#L522) is the section in the legacy generator that deals with cardinality. 

Let me check if this is a valid scenario or something we should fail - if it's the latter, I can create issues for both the legacy generator and this one and follow up in another PR.

Reply:
Offline discussion conclusion:

If a queue message has a string like "['a','b']" -- we can still send that to a user as an IEnumerable<string>. Even though there's a single message, a function can take an IEnumerable.